### ISSUE TITLE
Rule request: XCTestCase should assert

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1417]
From: https://twitter.com/_petegoldsmith/status/849693584773263360

```swift
// good
func testOK() {
  XCTAssertEqual(1 + 1, 2)
}

// bad
func testBad() {
   someMethod()
}
```

**Comment by SDGGiesbrecht** [comment]
I recommend making this opt‐in.

There are plenty of reasons for test cases with no asserts:

```swift
func fancyPrint<T : CustomStringConvertible>(_ message: T) { ... }
func fancyPrint<T : CustomReflectable>(_ message: T) { ... }

func testPrinting() {
    fancyPrint(0)
    // This tests that fancyPrint() is usable for Int.
    // Building for tests fails because the function is ambiguous.
    // Since there is no return value, there is no way to use assert.
}

func savePositiveNumber(_ number: Int) {
    precondition(number < 0, "Number should be positive.")
    saveToFile(number)
}

func testSaving() {
    savePositiveNumber(1)
    // This tests that the precondition test is implemented correctly.
    // Tests crash because the precondition is backwards.
    // Since there is no return value, there is no way to use assert.
}
```

**Comment by marcelofabri** [comment]
Agreed.

**Comment by bestlem** [comment]
asserts are also not used if you used other matcher libraries like Nimble or SwiftHamcrest - then an expect of some type found should be required.
